subworkflows/align_short/modules/nf-core/modules/samtools/merge/samtools-merge.diff

```diff
@@ -0,0 +1,14 @@
+Changes in module 'nf-core/modules/samtools/merge'
+--- modules/nf-core/modules/samtools/merge/main.nf
++++ modules/nf-core/modules/samtools/merge/main.nf
+@@ -25,7 +25,7 @@
+     script:
+     def args = task.ext.args   ?: ''
+     prefix   = task.ext.prefix ?: "${meta.id}"
+-    def file_type = input_files[0].getExtension()
++    def file_type = input_files[0].getExtension() ?: "bam"
```

Comment:
So the bam file in input for `samtools/merge`:
```
[[id:mMelMel3, datatype:hic], [/lustre/scratch123/tol/teams/tolit/users/ps22/pipelines/nf-core-pipeline/subworkflows/align_short/work/96/594101de5f106e8936f63727a10eab/mMelMel3_T1.sorted.bam]]
```
gets staged as `1`, not `0`, so without that extra `bam` in the code, the output file is `mMelMel3.` which throws an error.
```
work/
├── 81
│   └── 86a1c20b2e71bd47b301e08d8baf74
│       ├── 1
│       │   └── mMelMel3_T1.sorted.bam -> /lustre/scratch123/tol/teams/tolit/users/ps22/pipelines/nf-core-pipeline/subworkflows/align_short/work/96/594101de5f106e8936f63727a10eab/mMelMel3_T1.sorted.bam
│       ├── mMelMel3.bam
│       └── versions.yml
```

I think if I modify the `merge` similar to `view`, the issue may be more cleanly resolved. If you have suggestions to modify the input itself to comply better, let me know. I tried to follow their `test` example: https://github.com/nf-core/modules/blob/master/tests/modules/samtools/merge/main.nf

Reply:
OK. I could replicate the issue. I think it's a bug someone mentioned on Slack not so long ago. When the input of a module is a list of a single element, Nextflow automatically removes the list and gives the value directly. 
This means that `input_files[0]` is the first character of the path (i.e. `1`) rather than a full path.

There is a workaround that should be put in the module itself to check whether the input is a list, and wrap it back into a list otherwise. I'll try to find it and submit a pull-request on nf-core/modules/samtools/merge, and then we shouldn't need this patch any more

Reply:
Mahesh to the rescue - as always ? -
https://nfcore.slack.com/archives/CJRH30T6V/p1658229950275629

Reply:
Opened a pull-request: https://github.com/nf-core/modules/pull/2114

Reply:
The pull-request has been merged, so this subworkflow should now work with the default samtools/merge

Reply:
Updated